niceman/resource/base.py

```diff
@@ -73,141 +119,84 @@ def factory(config):
-        elif type_ and type_ in valid_resource_types:
-            try:
-                config = dict(cm.items(type_.split('-')[0]))
-            except NoSectionError:
-                config = {}
-        else:
-            type_ = ui.question(
-                "Enter a resource type",
-                # TODO: decision on type of a container, if needed
-                # needs to be done outside, and should be configurable
-                # or follow some heuristic (e.g. if name starts with a
-                # known type, e.g. docker-
-                default="docker-container"
-            )
+    def _find_resources(self, resref, resref_type):
+        def from_name(x):
+            return [(name, config) for name, config in self.inventory.items()
+                    if x == name]
+
+        def from_id(x):
+            return [(name, config) for name, config in self.inventory.items()
+                    if x == config.get("id")]
+
+        results_name = None
+        results_id = None
+        if resref_type == "auto":
+            results_name = from_name(resref)
+            results_id = from_id(resref)
+        elif resref_type == "name":
+            results_name = from_name(resref)
+        elif resref_type == "id":
+            results_id = from_id(resref)
+        return results_name, results_id
+
+    def _get_resource_config(self, resref, resref_type="auto"):
+        results_name, results_id = self._find_resources(resref, resref_type)
+        if results_name and results_id:
+            raise MultipleResourceMatches(
+                "{} is ambiguous. "
+                "Explicitly specify whether it is a name or id".format(resref))
+        elif not (results_name or results_id):
+            raise ResourceNotFoundError(
+                "Resource matching {} not found".format(resref))
+        elif results_id and len(results_id) > 1:
+            raise MultipleResourceMatches(
+                "ID {} matches multiple resources. "
+                "Try specifying with the name instead".format(resref))
+
+        name, inventory_config = (results_name or results_id)[0]
+        type_ = inventory_config['type']
+        try:
+            config = dict(self.config_manager.items(type_.split('-')[0]))
+        except NoSectionError:
             config = {}
-            if type_ not in valid_resource_types:
-                raise MissingConfigError(
-                    "Resource type '{}' is not valid".format(type_))
+        config.update(inventory_config)
+        return config
 
-        # Overwrite config settings with those from the command line.
-        config['name'] = name
-        if type_:
-            config['type'] = type_
-        if id_:
-            config['id'] = id_
-
-        return config, inventory
-
-    @staticmethod
-    def get_config_manager(config_path=None):
-        """Returns the information stored in the niceman.cfg file.
+    def get_resource(self, resref, resref_type="auto"):
```

Comment:
So, if I am reading this right, the default general configuration file (formerly known as niceman.cfg) is handled by the ConfigManager unless overridden with --config. What happens if --config is never passed at the command line?

Reply:
If --config isn't passed, ConfigManager looks for goes down its `/etc/niceman/niceman.cfg -> $XDG_CONFIG_DIRS -> $XDG_CONFIG_HOME -> $PWD/.niceman/niceman.cfg` processing, with settings from later files taking precedence.